src/analysis/function_parameters.rs

```diff
@@ -0,0 +1,235 @@
+use std::collections::HashMap;
+
+use config;
+use config::parameter_matchable::ParameterMatchable;
+use env::Env;
+use library;
+use nameutil;
+use super::conversion_type::ConversionType;
+use super::rust_type::rust_type;
+use super::ref_mode::RefMode;
+use super::out_parameters::can_as_return;
+use traits::IntoString;
+
+pub use config::functions::TransformationType;
+
+//TODO: remove unused fields
+#[derive(Clone, Debug)]
+pub struct RustParameter {
+    pub ind_c: usize, //index in `Vec<CParameter>`
+    pub name: String,
+    pub typ: library::TypeId,
+    pub allow_none: bool,
+}
```

Comment:
Its very strange, but now only needed in this struct is `ind_c` and I plan to remove others.